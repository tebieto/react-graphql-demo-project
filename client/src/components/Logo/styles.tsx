import styled from 'styled-components';

export const LogoContainer = styled.div`
  .wrapper {
    display: flex;
    align-items: center;
    cursor: pointer;
    .logo {
      height: 50px;

      @media screen and (max-width: 800px) {
        display: none;
      }
    }
  }
`;
