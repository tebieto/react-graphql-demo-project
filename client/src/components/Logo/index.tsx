import React from 'react';
import AppLogo from '../../assets/logo.svg';
import { LogoContainer } from './styles';
import { useHistory } from 'react-router';
import { PAGES } from '../../utils/constants';

const Logo = (): JSX.Element => {
  const history = useHistory();
  return (
    <LogoContainer>
      <div className="wrapper" onClick={() => history.push(PAGES.home)}>
        <img className="logo" src={AppLogo} alt="" />
      </div>
    </LogoContainer>
  );
};

export default Logo;
