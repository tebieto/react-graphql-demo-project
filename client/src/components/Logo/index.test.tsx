import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Logo from '.';

it('renders correctly', () => {
  const tree = shallow(<Logo />);
  expect(toJson(tree)).toMatchSnapshot();
});
