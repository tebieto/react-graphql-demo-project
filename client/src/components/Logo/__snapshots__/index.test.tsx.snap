// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <div
    className="wrapper"
    onClick={[Function]}
  >
    <img
      alt=""
      className="logo-mark"
      src="github-mark.png"
    />
    <img
      alt=""
      className="logo"
      src="github-logo.png"
    />
  </div>
</styled.div>
`;
